src/NServiceBus.Core/obsoletes-v8.cs

```diff
@@ -457,4 +463,76 @@ public class Scheduler
     }
 }
 
+namespace NServiceBus
+{
+    using DeliveryConstraints;
+
+    [ObsoleteEx(
+        Message = "Non-durable delivery support has been moved to the transports that can support it. See the upgrade guide for more details.",
+        TreatAsErrorFromVersion = "8",
+        RemoveInVersion = "9")]
+    public class NonDurableDelivery2 : DeliveryConstraint
+    {
+    }
+}
+
+namespace NServiceBus
+{
+    using Settings;
+    using System;
+
+    [ObsoleteEx(
+        Message = "Non-durable delivery support has been moved to the transports that can support it. See the upgrade guide for more details.",
+        TreatAsErrorFromVersion = "8",
+        RemoveInVersion = "9")]
+    public static class DurableMessagesConfig
+    {
+        public static void EnableDurableMessages(this EndpointConfiguration config)
+        {
+            throw new NotImplementedException();
```

Comment:
Do we have a guideline why NYI is thrown instead of Not Supported or this this replaced by ObsoleteEx weaving?

Reply:
I don't think it matters since this code will never run?

Reply:
It can be run when someone would overwrite the DLL on filesystem with a newer version.

Reply:
Seems like Fody throws an NotImplemented https://github.com/Fody/Obsolete#throwsnotimplementedtext so we can add any exception we want since fody will overwrite it